src/targets/registry.ts

```diff
@@ -59,18 +61,52 @@ export interface RegistryConfig {
   onlyIfPresent?: RegExp;
 }
 
+interface LocalRegistry {
+  dir: string;
+  git: SimpleGit;
+}
+
+interface ArtifactData {
+  url?: string;
+  checksums?: {
+    [key: string]: string;
+  };
+}
+
 /**
  * Target responsible for publishing static assets to GitHub pages
  */
 export class RegistryTarget extends BaseTarget {
+  /** A set of all created registry instances to be able to batch them */
+  private static instances: Set<RegistryTarget> = new Set();
+  // A promise-based lock to ensure non-parallel publishing when batching
+  // all registry targets. It is acquired and waited on at the publish stage
+  private static lock = new Mutex();
+  /** The information of the canonical local checkout of the registry */
+  private static localRepo: undefined | LocalRegistry;
   /** Target name */
-  public readonly name: string = 'registry';
+  public readonly name = 'registry';
   /** Target options */
   public readonly registryConfig: RegistryConfig;
   /** Github client */
   public readonly github: Github;
   /** Github repo configuration */
   public readonly githubRepo: GithubGlobalConfig;
+  private published = false;
+
+  public isPublished(): boolean {
+    return this.published;
+  }
```

Comment:
Why do we need this? Who is going to consume this outside of this class? We already access `this.published` directly below.

Reply:
Hrm, nevermind. In `isLastToPublish` we're dealing with sibling instances, not ourself.